src/cylinder.cpp

```diff
@@ -5,39 +5,4 @@
 
 #include "cylinder.h"
 
-VirtualCylinder* VirtualCylinder::virtualCylinder = new VirtualCylinder;
-
-int32_t Cylinder::getThingIndex(const Thing*) const { return -1; }
-
-size_t Cylinder::getFirstIndex() const { return 0; }
-
-size_t Cylinder::getLastIndex() const { return 0; }
-
-uint32_t Cylinder::getItemTypeCount(uint16_t, int32_t) const { return 0; }
-
-std::map<uint32_t, uint32_t>& Cylinder::getAllItemTypeCount(std::map<uint32_t, uint32_t>& countMap) const
-{
-	return countMap;
-}
-
-Thing* Cylinder::getThing(size_t) const { return nullptr; }
-
-void Cylinder::internalRemoveThing(Thing*)
-{
-	//
-}
-
-void Cylinder::internalAddThing(Thing*)
-{
-	//
-}
-
-void Cylinder::internalAddThing(uint32_t, Thing*)
-{
-	//
-}
-
-void Cylinder::startDecaying()
-{
-	//
-}
+std::shared_ptr<VirtualCylinder> VirtualCylinder::virtualCylinder = std::make_shared<VirtualCylinder>();
```

Comment:
The best approach here, would be to remove any variant of `Cylinder` and give it a name that better reflects its behavior :)

Could this be `const auto`
```suggestion
const auto VirtualCylinder::virtualCylinder = std::make_shared<VirtualCylinder>();
```